TweetCell.m

```diff
@@ -21,4 +14,70 @@ - (void)setSelected:(BOOL)selected animated:(BOOL)animated {
     // Configure the view for the selected state
 }
 
+- (IBAction)didTapFavorite:(UIButton *)sender {
+    if (!self.tweet.favorited) {
+        [[APIManager shared] favorite:self.tweet completion:^(Tweet *tweet, NSError *error) {
+            if(error){
+                 NSLog(@"Error favoriting tweet: %@", error.localizedDescription);
+            }
+            else{
+                NSLog(@"Successfully favorited the following Tweet: %@", tweet.text);
+                self.tweet.favorited = YES;
+                self.tweet.favoriteCount++;
+                [self refreshData]; // TODO: separate into refreshFavorites and refreshRetweets
+            }
+        }];
+    } else {
+        [[APIManager shared] unfavorite:self.tweet completion:^(Tweet *tweet, NSError *error) {
+            if(error){
+                 NSLog(@"Error unfavoriting tweet: %@", error.localizedDescription);
+            }
+            else{
+                NSLog(@"Successfully unfavorited the following Tweet: %@", tweet.text);
+                self.tweet.favorited = NO;
+                self.tweet.favoriteCount--;
+                [self refreshData];
+            }
+        }];
+    }
+    [self refreshData];
+}
+
+- (IBAction)didTapRetweet:(UIButton *)sender {
+    if (!self.tweet.retweeted) {
+        [[APIManager shared] retweet:self.tweet completion:^(Tweet *tweet, NSError *error) {
+            if(error){
+                 NSLog(@"Error retweeting tweet: %@", error.localizedDescription);
+            }
+            else{
+                NSLog(@"Successfully retweeted the following Tweet: %@", tweet.text);
+                self.tweet.retweeted = YES;
+                self.tweet.retweetCount++;
+                [self refreshData];
+            }
+        }];
+    } else {
+        [[APIManager shared] unretweet:self.tweet completion:^(Tweet *tweet, NSError *error) {
+            if(error){
+                 NSLog(@"Error unretweeting tweet: %@", error.localizedDescription);
+            }
+            else{
+                NSLog(@"Successfully unretweeted the following Tweet: %@", tweet.text);
+                self.tweet.retweeted = NO;
+                self.tweet.retweetCount--;
+                [self refreshData];
+            }
+        }];
+    }
+}
+
+- (void)refreshData {
```

Comment:
This name is confusing. Refresh is usually used for data that is new in some way (read from server or disk). This data is the same data, we're just decorating the cell. So I would name this `decorate` or `configureViews` or something.